builtin/credential/cert/path_login.go

```diff
@@ -236,7 +236,7 @@ func (b *backend) verifyCredentials(ctx context.Context, req *logical.Request, d
 	var certName string
 	if req.Auth != nil { // It's a renewal, use the saved certName
 		certName = req.Auth.Metadata["cert_name"]
-	} else {
+	} else if d != nil { // d is nil if handleAuthRenew call the authRenew
```

Comment:
Does the comment above not apply? I read the above as saying, if the request is a renewal (which this new comment seems to say is the case), we should be hitting the above.

I'm curious why `req.Auth` became nil _and_ `d` was nil? 

cc @sgmiller -- maybe you have more context as you were looking at cert auth fairly recently.

Reply:
It seems so. I figured to add this guard just in case. I am trying to add a go test to make sure though.

Reply:
I'd probably add another `else` in this case and ensure `certName` doesn't ever end up empty to guard against this occurring tbh.

Reply:
I think so, I was worried about that too but dug a bit deeper and found some handling for it.

Reply:
Ah, cool, thank you!

Reply:
I have added a test to practice token renewal. I could not trigger a code path with which both `req.Auth` and the data field are `nil`. And, the pathLoginRenew call contains req.Auth which is populated in the expiration manager [here](https://github.com/hashicorp/vault/blob/8b6953c2e7977e65bc83563664c5869b103bbef8/vault/expiration.go#L1976). I am happy with removing the extra logic here though having the guard would not hurt. Thoughts?